instantout/reservation/manager.go

```diff
@@ -0,0 +1,260 @@
+package reservation
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcutil"
+
+	reservationrpc "github.com/lightninglabs/loop/swapserverrpc"
+)
+
+// Manager manages the reservation state machines.
+type Manager struct {
+	// cfg contains all the services that the reservation manager needs to
+	// operate.
+	cfg *Config
+
+	// activeReservations contains all the active reservationsFSMs.
+	activeReservations map[ID]*FSM
+
+	// currentHeight stores the currently best known block height.
+	currentHeight int32
+
+	// blockEpochChan receives new block heights.
+	blockEpochChan chan int32
+
+	runCtx context.Context
```

Comment:
nit: there's no need for the `runCtx` as the context created inside `Run` could just be passed to all other functions.